app/src/main/java/com/nutomic/syncthingandroid/model/Completion.java

```diff
@@ -112,9 +108,12 @@ public void updateFromConfig(List<Device> newDevices, List<Folder> newFolders) {
     public int getDeviceCompletion(String deviceId) {
         int folderCount = 0;
         double sumCompletion = 0;
-        for (String folderId : deviceFolderMap.getOrDefault(deviceId, defaultDevice).keySet()) {
-            sumCompletion += (deviceFolderMap.get(deviceId)).get(folderId).completion;
-            folderCount++;
+        HashMap<String, CompletionInfo> folderMap = deviceFolderMap.get(deviceId);
+        if (folderMap != null) {
+            for (String folderId : folderMap.keySet()) {

```

Comment:
Iterate over values and just access completiom straight away without the need of double get...

Reply:
I have this already, uploading shortly ..